apis/Google.Storage.V1/Google.Storage.V1/SignedUrlUtils.cs

```diff
@@ -0,0 +1,410 @@
+        /// included with the request anyway if the desired content is encrypted with customer-supplied encryption keys.
+        /// </para>
+        /// <para>
+        /// If <paramref name="request"/> is null, no headers are included in the signed URL's signature, so any requests
+        /// made with the created URL must not contain Content-MD5, Content-Type, or any header started with "x-goog-".
+        /// </para>
+        /// <para>
+        /// Note that when GET is specified as the <paramref name="request"/>, both GET and HEAD requests can be made with
+        /// the created signed URL.
+        /// </para>
+        /// <para>
+        /// See https://cloud.google.com/storage/docs/access-control/signed-urls for more information on signed URLs.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        /// The <paramref name="credentialFilePath"/> does not refer to a valid JSON service account key file.
+        /// </exception>
+        /// <returns>
+        /// The signed URL which can be used to provide access to a bucket or object for a limited amount of time.
+        /// </returns>
+        public static string Create(
+            string bucket,
+            string objectName,
+            string credentialFilePath,
+            TimeSpan duration,
+            HttpRequestMessage request) =>
+                Create(bucket, objectName, credentialFilePath, DateTimeOffset.UtcNow + duration, request);
+
+        /// <summary>
+        /// Creates a signed URL which can be used to provide limited access to specific buckets and objects to anyone
+        /// in possession of the URL, regardless of whether they have a Google account.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. May be null, in which case the signed URL
+        /// will refer to the bucket instead of an object.</param>
+        /// <param name="credentialFilePath">The path to the JSON key file for a service account. Must not be null.</param>
+        /// <param name="expiration">The point in time after which the signed URL will be invalid. May be null, in which
+        /// case the signed URL never expires.</param>
+        /// <param name="request">A sample request for which the signed URL might be used. May be null.</param>
+        /// <remarks>
+        /// <para>
+        /// When the <paramref name="request"/> is specified, some of its headers will be included in the signed URL's
+        /// signature, and therefore must be included in requests made with the created URL. These are the Content-MD5 and
+        /// Content-Type content headers as well as any content or request header with a name starting with "x-goog-".
+        /// However, this does not apply to x-goog-encryption-key and x-goog-encryption-key-sha256 headers, but they must be
```

Comment:
Ok, I've removed these lines and now have a separate paragraph to explain this concept (as well as a link to the help docs on customer-supplied encryption keys). I'll merge after pushing again, but let me know if anything is still unclear and I'll address it in the next PR.